File: PHP/PedraPapelAtaqueAereoTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("PedraPapelAtaqueAereo.php");

class PedraPapelAtaqueAereoTest extends TestCase {

    public function testPedraPapelAtaqueAereo() {
        $obj = new PedraPapelAtaqueAereo();
        //Test 1
        $want = "Sem ganhador";
        $got = $obj->jokenpoWinner("pedra", "pedra");
        $this->assertEquals($want, $got);
        //Test 2
        $want = "Jogador 1 venceu";
        $got = $obj->jokenpoWinner("ataque", "papel");
        $this->assertEquals($want, $got);
        //Test 3
        $want = "Jogador 1 venceu";
        $got = $obj->jokenpoWinner("Ataque", "Pedra");
        $this->assertEquals($want, $got);
        //Test 4
        $want = "Jogador 2 venceu";
        $got = $obj->jokenpoWinner("Pedra", "Ataque");
        $this->assertEquals($want, $got);
        //Test 5
        $want = "Jogador 1 venceu";
        $got = $obj->jokenpoWinner("Pedra", "Papel");
        $this->assertEquals($want, $got);
        //Test 6
        $want = "Jogador 2 venceu";
        $got = $obj->jokenpoWinner("Papel", "Pedra");
        $this->assertEquals($want, $got);
        //Test 7
        $want = "Jogador 1 venceu";
        $got = $obj->jokenpoWinner("Ataque", "Papel");
        $this->assertEquals($want, $got);
        //Test 8
        $want = "Jogador 2 venceu";
        $got = $obj->jokenpoWinner("Papel", "Ataque");
        $this->assertEquals($want, $got);
        //Test 9
        $want = "Ambos venceram";
        $got = $obj->jokenpoWinner("Papel", "Papel");
        $this->assertEquals($want, $got);
        //Test 10
        $want = "Sem ganhador";
        $got = $obj->jokenpoWinner("Pedra", "Pedra");
        $this->assertEquals($want, $got);
        //Test 11
        $want = "Aniquilacao mutua";
        $got = $obj->jokenpoWinner("Ataque", "Ataque");
        $this->assertEquals($want, $got);
    }

}

File: GO/PedraPapelAtaqueAereo_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"testing"
)

func TestPedraPapelAtaqueAereo(t *testing.T) {
	tables := []struct {
		p1   string
		p2   string
		want string
	}{
		//Test 1
		{"pedra", "pedra", "Sem ganhador"},
		//Test 2
		{"ataque", "papel", "Jogador 1 venceu"},
		//Test 3
		{"Ataque", "Pedra", "Jogador 1 venceu"},
		//Test 4
		{"Pedra", "Ataque", "Jogador 2 venceu"},
		//Test 5
		{"Pedra", "Papel", "Jogador 1 venceu"},
		//Test 6
		{"Papel", "Pedra", "Jogador 2 venceu"},
		//Test 7
		{"Ataque", "Papel", "Jogador 1 venceu"},
		//Test 8
		{"Papel", "Ataque", "Jogador 2 venceu"},
		//Test 9
		{"Papel", "Papel", "Ambos venceram"},
		//Test 10
		{"Pedra", "Pedra", "Sem ganhador"},
		//Test 11
		{"Ataque", "Ataque", "Aniquilacao mutua"},
	}

	for _, table := range tables {
		got := jokenpoWinner(table.p1, table.p2)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}
